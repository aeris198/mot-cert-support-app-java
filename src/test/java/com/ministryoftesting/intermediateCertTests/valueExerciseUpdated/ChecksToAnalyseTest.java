package com.ministryoftesting.intermediateCertTests.valueExerciseUpdated;

import com.ministryoftesting.db.UserDB;
import com.ministryoftesting.models.user.User;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.sql.SQLException;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

public class ChecksToAnalyseTest {

    private static UserDB userDB;

    @BeforeAll
    public static void setup() {
        userDB = new UserDB();
    }

    @Test
    public void checkClassCanBeCreated(){
        assertEquals(UserDB.class, userDB.getClass());
    }

    @Test
    public void userCanBeCreated() throws SQLException {
        User user = new User("Jon", "test@email.com", "password", "user");
        User result = userDB.createUser(user);

        assertEquals(User.class, result.getClass());
    }

    @Test
    public void returnsPositiveResultWhenDeletingUser() throws SQLException {
        boolean result = userDB.deleteUser(1);

        assertTrue(result);
    }

    @Test
    public void returnsNegativeResultWhenDeletingNonUser() throws SQLException {
        boolean result = userDB.deleteUser(122);

        assertFalse(result);
    }

    @Test
    public void testUserList() throws SQLException {

        //cleardown any existing projects in the DB
        for (int i = userDB.getUsers().size(); i > 0; i--) {
            userDB.deleteUser(i);
        }

        User user = new User("Jon", "test@email.com", "password", "user");
        userDB.createUser(user);
        User user2 = new User("Jon", "test@email.com", "password", "user");
        userDB.createUser(user);
        User user3 = new User("Jon", "test@email.com", "password", "user");
        userDB.createUser(user);

        List<User> users = userDB.getUsers();

        Approvals.verify(users.toString());
    }

}
