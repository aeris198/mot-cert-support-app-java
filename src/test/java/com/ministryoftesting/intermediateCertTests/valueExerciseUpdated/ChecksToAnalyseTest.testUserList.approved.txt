[User{id=3, username='Jon', email='test@email.com', password='password', role='user'}, User{id=4, username='Jon', email='test@email.com', password='password', role='user'}, User{id=5, username='Jon', email='test@email.com', password='password', role='user'}]